File: src/main/java/com/codeperfection/shipit/dto/shipping/CreateShippingDto.java
package com.codeperfection.shipit.dto.shipping;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;
import java.time.ZoneId;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CreateShippingDto {

    @NotEmpty
    @Size(max = 256)
    private String name;

    @NotNull
    private UUID transporterUuid;

    @NotNull
    private ZoneId timeZoneName;
}

File: src/main/kotlin/com/codeperfection/shipit/dto/shipping/CreateShippingDto.kt
package com.codeperfection.shipit.dto.shipping

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size
import java.util.UUID

data class CreateShippingDto(
    @field:NotBlank
    @field:Size(max = 256)
    val name: String,
    val transporterId: UUID
)
